TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -391,5 +391,31 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
             var savepointName = context.Diagnostics == null ? new SymbolReference(new AlphanumericValue((Token)context.savepoint_name), SymbolType.SqlIdentifier) : null;
             return new ReleaseSavepointStatement(savepointName);
         }
+
+        public ExecuteImmediateStatement CreateExecuteImmediateStatement(CodeElementsParser.ExecuteImmediateStatementContext context)
+        {
+            if (context.sqlExpression() == null) return null;
+            if (context.sqlExpression().sqlVariable() != null)
+            {
+                return new ExecuteImmediateStatement(CreateSqlVariable(context.sqlExpression().sqlVariable()));
+            }
+            else if (context.sqlExpression().sqlConstant() != null)
+            {
+                var stringLiteral =
+                    new SqlConstant(ParseTreeUtils.GetFirstToken(context.sqlExpression().sqlConstant()));
+                //TODO add support for string expressions
+            }
```

Comment:
Why separate the cases ? Execute immediate is made of a single expression, you can create the statement without checking the nature of the expression.
However you have to check the nature of the expression after the statement has been created in a dedicated checker

Reply:
I corrected that by  creating the statement without checking the nature of the expression.
But i didn't understand why i should  check  the nature of the expression after the statement has been created in a dedicated checker.

Reply:
This is the way the parser works. Builder's role is to create all objects reflecting what has been written. What is written may be wrong and depending on the grammar, errors may be detected directly by ANTLR or later by a checker.
In this case, the grammar allows all expressions after EXECUTE IMMMEDIATE but the specs says that only variables and string-expressions are allowed. That's why we need a dedicated checker to ensure that the expression found is valid in this context.

Furthermore, as we don't support string-expressions yet, the checker will allow only 2 forms of expressions:
- variable
- alphanumeric constant (use ExpressionType and ConstantType)